src/ORM/Association.php

```diff
@@ -815,6 +815,11 @@ protected function _formatAssociationResults($query, $surrogate, $options)
         $formatters = $surrogate->formatResults();
 
         if (!$formatters || empty($options['propertyPath'])) {
+            if ($formatters) {
```

Comment:
These nested conditionals are a bit odd, as `if($formatters)` is inside `if(!$formatters)`. Perhaps the new `if($formatters)` case can be hoisted above this conditional?